packages/react-scripts/scripts/init.js

```diff
@@ -76,6 +82,16 @@ module.exports = function(appPath, appName, verbose, originalDirectory) {
   }
   args.push('react', 'react-dom');
 
+  // Install additional template dependencies, if present
+  var templateDependenciesPath = path.join(appPath, '.template.dependencies.json');
```

Comment:
>The main problem I find in this hole situation is that we're adding the --template cli option, that opens ways to create a custom template, but (for now^) should be only for internal usage.

Maybe we can keep it undocumented and making it uglier, e.g. `--internal-testing-template`.